src/sync/index.js

```diff
@@ -83,259 +84,23 @@ async function sync(db){
     var endBlock = Math.min(startBlock + batchSize - 1, currentBlockChainHeight);
     var syncTopic = false, syncCOracle = false, syncDOracle = false, syncVote = false, syncOracleResult = false, syncFinalResult = false;
 
-    // sync TopicCreated
-    var syncTopicCreatedPromise = new Promise(async (resolve) => {
-      let result;
-      try {
-        result = await qclient.searchLogs(startBlock, endBlock, Contracts.EventFactory.address, [Contracts.EventFactory.TopicCreated], Contracts, removeHexPrefix);
-        console.log('searchlog TopicCreated')
-      } catch(err) {
-        console.error(`Error: ${err.message}`);
-        resolve();
-        return;
-      }
-
-      console.log(`${startBlock} - ${endBlock}: Retrieved ${result.length} entries from TopicCreated`);
-      var createTopicPromises = [];
-
-      _.forEach(result, (event, index) => {
-        let blockNum = event.blockNumber;
-        let txid = event.transactionHash;
-        _.forEachRight(event.log, (rawLog) => {
-          if(rawLog['_eventName'] === 'TopicCreated'){
-            var insertTopicDB = new Promise(async (resolve) =>{
-              try {
-                var topic = new Topic(blockNum, txid, rawLog).translate();
-                await db.Topics.insert(topic);
-                resolve();
-              } catch(err) {
-                console.error(`Error: ${err.message}`);
-                resolve();
-                return;
-              }
-            });
-
-            createTopicPromises.push(insertTopicDB);
-          }
-        });
-      });
-
-      Promise.all(createTopicPromises).then(()=>{
-        resolve()
-      });
+    var syncTopicCreatedPromise = new Promise(async (resolve) => { 
```

Comment:
I prefer to delay this refactor as it requires manual testing. I dunt refactor it before because
1. func wont be reused, only mk it cleaner
2. multi insert db optimization should be done first

Reply:
as discussed, going to parse TopicCreated before Oracle parsing to ensure topic is available to populate oracle info.